compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14572,6 +14573,45 @@ private STNode parseXMLNamePatternChain(STNode startToken) {
         return STNodeFactory.createXMLNamePatternChainingNode(startToken, xmlNamePattern, gtToken);
     }
 
+    /**
+     * Parse xml step extend chain.
+     * <p>
+     * <code>
+     * xml-step-extend := .< xml-name-pattern > | [ expression ] | .method-name ( arg-list )
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseXMLStepExtend() {
+        List<STNode> xmlStepExtendList = new ArrayList<>();
+        STToken nextToken = peek();
+
+        if (isEndOfXMLStepExtend(nextToken.kind)) {
+            return STNodeFactory.createNodeList(xmlStepExtendList);
+        }
+
+        startContext(ParserRuleContext.XML_STEP_EXTEND);
+        STNode stepExtension;
+        while (!isEndOfXMLStepExtend(nextToken.kind)) {
+            if (nextToken.kind == SyntaxKind.DOT_TOKEN) {
```

Comment:
It would be cleaner to extract `xml-step-extend` grammar parsing to a separate method. (we will also have the flexibility to reuse it.) We can call the current method `parseXMLStepExtends()` or `parseXMLStepExtendList()`.

doc grammer: xml-step-extends := xml-step-extend*